File: SDKs/amazonaws.com/mturk-requester/2017-01-17/go/pkg/models/operations/deleteworkerblock.go
package operations

import (
	"openapi/pkg/models/shared"
)

type DeleteWorkerBlockXAmzTargetEnum string

const (
	DeleteWorkerBlockXAmzTargetEnumMTurkRequesterServiceV20170117DeleteWorkerBlock DeleteWorkerBlockXAmzTargetEnum = "MTurkRequesterServiceV20170117.DeleteWorkerBlock"
)

type DeleteWorkerBlockHeaders struct {
	XAmzAlgorithm     *string                         `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                         `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                         `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                         `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                         `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                         `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                         `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        DeleteWorkerBlockXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type DeleteWorkerBlockRequest struct {
	Headers DeleteWorkerBlockHeaders
	Request shared.DeleteWorkerBlockRequest `request:"mediaType=application/json"`
}

type DeleteWorkerBlockResponse struct {
	ContentType               string
	DeleteWorkerBlockResponse map[string]interface{}
	RequestError              *interface{}
	ServiceFault              *interface{}
	StatusCode                int64
}

File: SDKs/amazonaws.com/mturk-requester/2017-01-17/typescript/src/sdk/models/operations/deleteworkerblock.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum DeleteWorkerBlockXAmzTargetEnum {
    MTurkRequesterServiceV20170117DeleteWorkerBlock = "MTurkRequesterServiceV20170117.DeleteWorkerBlock"
}


export class DeleteWorkerBlockHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: DeleteWorkerBlockXAmzTargetEnum;
}


export class DeleteWorkerBlockRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: DeleteWorkerBlockHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.DeleteWorkerBlockRequest;
}


export class DeleteWorkerBlockResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  deleteWorkerBlockResponse?: Map<string, any>;

  @SpeakeasyMetadata()
  requestError?: any;

  @SpeakeasyMetadata()
  serviceFault?: any;

  @SpeakeasyMetadata()
  statusCode: number;
}
